cpp/src/sort/stable_sort.cu

```diff
@@ -37,6 +37,28 @@ std::unique_ptr<column> stable_sorted_order(table_view const& input,
   return sorted_order<true>(input, column_order, null_precedence, stream, mr);
 }
 
+std::unique_ptr<table> stable_sort(table_view const& input,
+                                   std::vector<order> const& column_order,
+                                   std::vector<null_order> const& null_precedence,
+                                   rmm::cuda_stream_view stream,
+                                   rmm::mr::device_memory_resource* mr)
+{
+  // fast-path sort conditions: single, non-floating-point, fixed-width column with no nulls
+  if (input.num_columns() == 1 && !input.column(0).has_nulls() &&
```

Comment:
Done in `"common_sort_impl.cuh"`. I also took the opportunity to unify all templated sort calls as taking a `sort_method` enum template parameter, rather than a `bool` to indicate stable sort.